app/bucketlist/views.py

```diff
@@ -0,0 +1,130 @@
+from app.models import User, BucketList, Item
+from flask import request, jsonify, Blueprint
+from app import db, jwt
+from flask_jwt import jwt_required, current_identity
+from app.auth.validate import validate_json
+from datetime import datetime
+from sqlalchemy.sql import func
+
+
+bucketlists = Blueprint('bucketlists', __name__,
+                        url_prefix='/api/v1/bucketlists')
+
+
+# Get all bucketlists and search bucketlists
+@bucketlists.route('/', methods=['GET'])
+@jwt_required()
+def get_bucketlist():
+    search = request.args.get('q', '')
+    page = request.args.get('page', 1)
+    limit = request.args.get('limit', 20)
+
+    results = BucketList.query.filter_by(user_id=current_identity.id).filter(
+        BucketList.name.ilike(f'%{search}%')).paginate(int(page), int(limit), False)
+    if not results:
+        return jsonify({"error": "BucketList not found"}), 404
+
+    buckets = [bucket.to_json() for bucket in results.items]
+
+    return jsonify(buckets), 200
+
+
+# Add new bucketlist
+@bucketlists.route('/', methods=['POST'])
+@jwt_required()
+@validate_json('name')
+def new_bucketlist():
+    data = request.get_json()
+
+    bucket = BucketList.query.filter_by(name=data['name']).first()
```

Comment:
What if the bucketlist belongs to another user. Give two user the liberty of having a bucketlist with the same name. Something like this will work for this: `bucket = BucketList.query.filter_by(name=data['name']).filter_by(owner=current_user_id).first()` which checks only bucketlists by current logged in user.